### DISCUSSION TITLE
Desired isolation level for team members

### DISCUSSION THREAD

**Discussion by Cu3PO42** [robocode-dev/tank-royale#112] [Q&A] [answered]
Hi there,

I understand that under the new paradigm of Tank Royale it's difficult to restrict what bot processes can and cannot do. But I'm wondering what the standard is that "well-behaved" teams should adhere to. I see that team messages are part of the standard communication, so I assume that is the intended communication mechanism. However, multiple processes running on the same host could also use other mechanisms for IPC. Is it correct to say that they should not do that? 

I'd appreciate some clarification so I can try to enforce the correct isolation in a sandbox. 

Thank you!

**Comment by flemming-n-larsen** [comment, marked as answer]
@Cu3PO42 
A bot and team member is allowed to use any framework they want beside that from Robocode. 🙂 
And there is no security manager in Tank Royale as bots run in their own process and not inside the same JVM.

So If you want to use another IPC process that should be okay. Just keep in mind that if the framework you use is not supported or require special setup on the system, it will be difficult to share the bot(s) and run those anywhere, as only plain JVM or .Net is supported with the current Bot APIs.

**Reply by Cu3PO42** [reply]
Thanks for the swift reply!

I'll take a that into account and not isolate bots on the same team from each other. (Context: I'm working on isolating participating bots from each other and the server, so a malicious bot cannot easily manipulate a battle.) I understand the difficulties of distributing bots with additional dependencies, but for me that's a separate problem to be solved :-)
